examples/kalman_filter.py

```diff
@@ -0,0 +1,374 @@
+r"""Illustrate how a Kalman-like filter can be defined on Lie groups.
+
+A generic Kalman filter class is defined for systems on Lie groups for which
+the exponential is subjective. Its use is illustrated on two localization
+problems, a linear and a non-linear one. In both cases, the propagation model
+is known, and sparse position measurements are obtained. It thus relies on a
+`model` of the system, providing the system's equations and jacobians.
+
+The former is a 1D-localization problem where the state is a 2D vector (x, v)
+made of the system's position and speed. The process writes
+(x_{i+1}, v_{i+1}) = (x_i + dt * v_i, v_i + dt * a_i),
+where dt is the time-step between i and i+1, and a_i a noisy acceleration
+measured by a given sensor.
+
+The latter is a 2D pose (position + orientation) estimation problem, where the
+state (R, x) is made a planar rotation and 2D position, i.e. a member of SE(2).
+The non-linear propagation writes
+:math:`(R_{i+1}, x_{i+1}) = (R_i \Omega_i, x_i + dt * R_i u_i)`,
+where :math:`\Omega_i, u_i` is the measured odometry of the system.
+The implementation follows that of the Invariant Extended Kalman Filter (IEKF)
+which was designed for such cases, see A. Barrau, S. Bonnabel, The Invariant
+Extended Kalman Filter as a Stable Observer, IEEE Transactions on Automatic
+Control, 2017.
+"""
+
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import geomstats.backend as gs
+from geomstats import algebra_utils
+from geomstats.geometry.euclidean import Euclidean
+from geomstats.geometry.special_euclidean import SpecialEuclidean
+
+
+class LocalizationLinear:
+    """Class for modeling a linear 1D localization problem.
+
+    The state is made of a scalar position and scalar speed, thus a 2D vector.
+    A sensor provides acceleration inputs, while another one provides sparse
+    measurements of the position.
+    """
+    def __init__(self):
+        self.group = Euclidean(2)
```

Comment:
It seems that this class doens't have any state variable, these look all like static variables and don't need to be initialized in init. Then most of the methods could be classmethods.